api/app/Services/Zoom/ZoomService.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace App\Services\Zoom;
+
+use App\Repositories\User\UserRepository;
+use Illuminate\Support\Facades\Http;
+
+final class ZoomService
+{
+    private const ZOOM_MEETING_API_ENDPOINT = 'https://api.zoom.us/v2/users/me/meetings';
+    private const ZOOM_GET_USER_API_ENDPOINT = 'https://api.zoom.us/v2/users/me';
+    private const ZOOM_OAUTH_TOKEN_POST_REQUEST_URL = 'https://zoom.us/oauth/token';
+
+    private $userRepository;
+
+    public function __construct(UserRepository $userRepository)
+    {
+        $this->userRepository = $userRepository;
+    }
+
+    public function meeting($event)
+    {
+        $user = $this->userRepository->getById($event->eventType->owner_id);
+
+        $response = Http::withHeaders([
+            "Authorization" => "Bearer " . $user->zoom_access_token,
+        ])->post(self::ZOOM_MEETING_API_ENDPOINT, [
+            "topic" => $event->eventType->name,
+            "start_time" => $event->start_date,
+            "timezone" => $event->timezone
+        ]);
+
+        $zoomResponseArr = json_decode($response->body(), true);
+
+        $event->zoom_meeting_link = $zoomResponseArr['start_url'];
+        $event->save();
+    }
+
+    public function saveToken($request)
```

Comment:
Try to pass only the needed data for the method instead of the whole request instance